synapse/storage/database.py

```diff
@@ -247,6 +301,12 @@ def _do_execute(self, func, sql: str, *args: Any) -> None:
     def close(self) -> None:
         self.txn.close()
 
+    def __enter__(self) -> "LoggingTransaction":
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        self.close()
```

Comment:
Did we never previously used `LoggingTransaction` as a context manager even though `Transaction` can be used that way?

Reply:
Exactly.